test/fluree/db/branch_test.clj

```diff
@@ -0,0 +1,280 @@
+            (is (= ["Alice"] results) "Should find Alice in main branch")))
+
+        (testing "Create feature branch from main"
+          (let [branch-result @(fluree/create-branch! conn "test-ledger:feature-branch" "test-ledger:main")]
+            (is (= "feature-branch" (:name branch-result)))
+            (is (= "main" (get-in branch-result [:created-from :branch])))
+            (is (some? (:created-at branch-result)))
+            (is (some? (:head branch-result))))
+
+          (testing "Add data to feature branch"
+            @(fluree/insert! conn "test-ledger:feature-branch"
+                             {"@context" {"ex" "http://example.org/"}
+                              "@graph" [{"@id" "ex:bob"
+                                         "@type" "Person"
+                                         "ex:name" "Bob"}]})
+
+            (testing "Verify branch isolation"
+              ;; Main branch should only have Alice
+              (let [main-db @(fluree/load conn "test-ledger:main")
+                    main-results @(fluree/query main-db {"@context" {"ex" "http://example.org/"}
+                                                         "select" "?name"
+                                                         "where" {"@id" "?id" "ex:name" "?name"}})]
+                (is (= ["Alice"] main-results) "Main branch should only have Alice"))
+
+              ;; Feature branch should have both Alice and Bob
+              (let [feature-db @(fluree/load conn "test-ledger:feature-branch")
+                    feature-results @(fluree/query feature-db {"@context" {"ex" "http://example.org/"}
+                                                               "select" "?name"
+                                                               "where" {"@id" "?id" "ex:name" "?name"}})]
+                (is (= (set feature-results) #{"Alice" "Bob"})
+                    "Feature branch should have both Alice and Bob"))))
+
+          (testing "Branch info and metadata"
+            (let [main-info @(fluree/branch-info conn "test-ledger:main")
+                  feature-info @(fluree/branch-info conn "test-ledger:feature-branch")]
+              (is (= "main" (:name main-info)))
+              (is (some? feature-info) "feature-info should not be nil")
+              (when feature-info
+                (is (= "feature-branch" (:name feature-info)))
+                (is (some? (:head feature-info)))
+                (is (some? (:created-at feature-info)))
+                (is (= "main" (get-in feature-info [:created-from "f:branch"]))))))
+
+          (testing "Branch deletion"
+            @(fluree/delete-branch! conn "test-ledger:feature-branch")
+            ;; Verify branch is deleted by attempting to load it
+            (let [result @(fluree/load conn "test-ledger:feature-branch")]
+              (is (util/exception? result) "Should return an exception when loading deleted branch")
```

Comment:
Do we get a pretty error here? I'd prefer to at least extract the message and test for it explicitly.